src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/induction/InPrisonInterestsResourceMapper.kt

```diff
@@ -67,11 +67,11 @@ class InPrisonInterestsResourceMapper(
       InPrisonTrainingInterest(trainingType = trainingType, trainingTypeOther = trainingTypeOther)
     } ?: emptyList()
 
-  private fun toInPrisonWorkTypes(workInterests: List<InPrisonWorkInterest>): Set<InPrisonWorkType> =
-    workInterests.map { InPrisonWorkType.valueOf(it.workType.name) }.toSet()
+  private fun toInPrisonWorkTypes(workInterests: List<InPrisonWorkInterest>): Set<InPrisonWorkType>? =
+    workInterests.map { InPrisonWorkType.valueOf(it.workType.name) }.toSet().ifEmpty { null }
```

Comment:
I quite like that `ifEmpty` syntax; it makes for some pretty neat concise readable code 👍 
I'm so glad we discovered it (by "we discovered" or course I mean intellij showing us the error of our ways! 😁 )